sycl/source/handler.cpp

```diff
@@ -1722,9 +1722,9 @@ handler::getContextImplPtr() const {
   return MQueue->getContextImplPtr();
 }
 
-void handler::setKernelCacheConfig(
-    sycl::detail::pi::PiKernelCacheConfig Config) {
-  impl->MKernelCacheConfig = Config;
+void handler::setKernelCacheConfig(int32_t Config) {
```

Comment:
I would add `static_assert` here that `int32_t` is huge enough to represent values from `PiKernelCacheConfig`. If the latter is `enum`, then we can even use [`std::underlying_type`](https://en.cppreference.com/w/cpp/types/underlying_type).

I suppose that compiler is capable of emitting a warning for `static_cast` whose result can't fit into a target type, but we do such cast in headers and warnings are suppressed in there, so we won't know if `PI_*` is suddenly bigger than 32-bit and being truncated.